TaskLayer/SearchTask/SearchTask.cs

```diff
@@ -169,37 +169,50 @@ protected override MyTaskResults RunSpecific(string OutputFolder, List<DbForTask
                 // nonspecific search
                 else if (SearchParameters.SearchType == SearchType.NonSpecific)
                 {
-                    for (int currentPartition = 0; currentPartition < combinedParams.TotalPartitions; currentPartition++)
+                    List<CommonParameters> paramsToUse = new List<CommonParameters> { combinedParams };
```

Comment:
Why are you making paramsToUse instead of just using combinedParams, especially if combined params is the only thing within the paramsToUse list?

Reply:
For semi, I need to search two different params: one for N-terminal specific cleavages and one for C-terminal. If it's semi, the list gets modified (see next line).

Reply:
Ok. I understand now